Usage tracker web scale (#10844)
grafana/mimir: docs/sources/mimir/configure/configuration-parameters/index.md

  # CLI flag: -usage-tracker.enabled
  [enabled: <boolean> | default = false]

  # Number of partitions to use for the usage-tracker. This number isn't

>   # expected to change once usage-tracker is already being used.

Reviewer: ```suggestion
  # expected to change after you're already using the usage-tracker.
```

  # CLI flag: -usage-tracker.partitions
  [partitions: <int> | default = 64]

  # Interval to reconcile partitions.
  # CLI flag: -usage-tracker.partition-reconcile-interval
  [partition_reconcile_interval: <duration> | default = 10s]

>   # Time to wait beforeshutting down a partition that is no longer owned by this

Reviewer: ```suggestion
  # Time to wait before shutting down a partition that is no longer owned by this
```

      # and
      # -usage-tracker.events-storage.writeringest-storage.kafka.max-consumer-lag-at-startup
      # to 0 to disable waiting for maximum consumer lag being honored at
      # startup.
      # CLI flag: -usage-tracker.events-storage.writeringest-storage.kafka.target-consumer-lag-at-startup
      [target_consumer_lag_at_startup: <duration> | default = 2s]

      # The guaranteed maximum lag before a consumer is considered to have
      # caught up reading from a partition at startup, becomes ACTIVE in the
      # hash ring and passes the readiness check. Set both
      # -usage-tracker.events-storage.writeringest-storage.kafka.target-consumer-lag-at-startup
      # and
      # -usage-tracker.events-storage.writeringest-storage.kafka.max-consumer-lag-at-startup
      # to 0 to disable waiting for maximum consumer lag being honored at
      # startup.
      # CLI flag: -usage-tracker.events-storage.writeringest-storage.kafka.max-consumer-lag-at-startup
      [max_consumer_lag_at_startup: <duration> | default = 15s]

      # Enable auto-creation of Kafka topic on startup if it doesn't exist. If

>       # creating the topic fails and the topic doesn't already exist, Mimir will
>       # fail to start.

Reviewer: ```suggestion
      # creating the topic fails and the topic doesn't already exist, Mimir
      # fails to start.
```

      # CLI flag: -usage-tracker.events-storage.writerauto-create-topic-enabled
      [auto_create_topic_enabled: <boolean> | default = true]

      # When auto-creation of Kafka topic is enabled and this value is positive,

>       # Mimir will create the topic with this number of partitions. When the

Reviewer: ```suggestion
      # Mimir creates the topic with this number of partitions. When the
```

>       # value is -1 the Kafka broker will use the default number of partitions

Reviewer: ```suggestion
      # value is -1, the Kafka broker uses the default number of partitions
```

      # unless for testing purposes.
      # CLI flag: -usage-tracker.events-storage.writerproducer-max-record-size-bytes
      [producer_max_record_size_bytes: <int> | default = 15983616]

      # The maximum size of (uncompressed) buffered and unacknowledged produced
      # records sent to Kafka. The produce request fails once this limit is
      # reached. This limit is per Kafka client. 0 to disable the limit.
      # CLI flag: -usage-tracker.events-storage.writerproducer-max-buffered-bytes
      [producer_max_buffered_bytes: <int> | default = 1073741824]

      # The maximum allowed for a read requests processed by an ingester to wait
      # until strong read consistency is enforced. 0 to disable the timeout.
      # CLI flag: -usage-tracker.events-storage.writerwait-strong-read-consistency-timeout
      [wait_strong_read_consistency_timeout: <duration> | default = 20s]

      # The maximum number of concurrent fetch requests that the ingester makes
      # when reading data from Kafka during startup. Concurrent fetch requests
      # are issued only when there is sufficient backlog of records to consume.

>       # 0 to disable.

Reviewer: ```suggestion
      # Set to 0 to disable.
```

      # and
      # -usage-tracker.events-storage.readeringest-storage.kafka.max-consumer-lag-at-startup
      # to 0 to disable waiting for maximum consumer lag being honored at
      # startup.
      # CLI flag: -usage-tracker.events-storage.readeringest-storage.kafka.target-consumer-lag-at-startup
      [target_consumer_lag_at_startup: <duration> | default = 2s]

      # The guaranteed maximum lag before a consumer is considered to have
      # caught up reading from a partition at startup, becomes ACTIVE in the
      # hash ring and passes the readiness check. Set both
      # -usage-tracker.events-storage.readeringest-storage.kafka.target-consumer-lag-at-startup
      # and
      # -usage-tracker.events-storage.readeringest-storage.kafka.max-consumer-lag-at-startup
      # to 0 to disable waiting for maximum consumer lag being honored at
      # startup.
      # CLI flag: -usage-tracker.events-storage.readeringest-storage.kafka.max-consumer-lag-at-startup
      [max_consumer_lag_at_startup: <duration> | default = 15s]

      # Enable auto-creation of Kafka topic on startup if it doesn't exist. If

>       # creating the topic fails and the topic doesn't already exist, Mimir will
>       # fail to start.

Reviewer: ```suggestion
      # creating the topic fails and the topic doesn't already exist, Mimir
      # fails to start.
```

      # CLI flag: -usage-tracker.events-storage.readerauto-create-topic-enabled
      [auto_create_topic_enabled: <boolean> | default = true]

      # When auto-creation of Kafka topic is enabled and this value is positive,

>       # Mimir will create the topic with this number of partitions. When the

Reviewer: ```suggestion
      # Mimir creates the topic with this number of partitions. When the
```

>       # value is -1 the Kafka broker will use the default number of partitions

Reviewer: ```suggestion
      # value is -1, the Kafka broker uses the default number of partitions
```

      # unless for testing purposes.
      # CLI flag: -usage-tracker.events-storage.readerproducer-max-record-size-bytes
      [producer_max_record_size_bytes: <int> | default = 15983616]

      # The maximum size of (uncompressed) buffered and unacknowledged produced
      # records sent to Kafka. The produce request fails once this limit is
      # reached. This limit is per Kafka client. 0 to disable the limit.
      # CLI flag: -usage-tracker.events-storage.readerproducer-max-buffered-bytes
      [producer_max_buffered_bytes: <int> | default = 1073741824]

      # The maximum allowed for a read requests processed by an ingester to wait
      # until strong read consistency is enforced. 0 to disable the timeout.
      # CLI flag: -usage-tracker.events-storage.readerwait-strong-read-consistency-timeout
      [wait_strong_read_consistency_timeout: <duration> | default = 20s]

      # The maximum number of concurrent fetch requests that the ingester makes
      # when reading data from Kafka during startup. Concurrent fetch requests
      # are issued only when there is sufficient backlog of records to consume.

>       # 0 to disable.

Reviewer: ```suggestion
      # Set to 0 to disable.
```
